### ISSUE TITLE
enlighten deftest forms

### ISSUE THREAD

**Issue by ryfow** [clojure-emacs/cider-nrepl#382]
Enlighten mode doesn't work with `deftest` forms because `deftest` puts the test code in the var's metadata and puts a call to `test-var` in the fn body.

I think this can be resolved with something like [what the debugger does](https://github.com/clojure-emacs/cider-nrepl/blob/master/src/cider/nrepl/middleware/util/instrument.clj#L74) but I can't say I really understand what's going on there.

## Expected behavior

`deftest` forms that are evalled with `enlighten-mode` on would show enlighten info when run.

## Actual behavior

enlighten info is not shown

### Step 1: Write some code like this:

```clojure
(require '[clojure.test :refer :all])

(defn not-a-test []
  (let [a (System/currentTimeMillis)]
    (is (= 0 (mod a 3)))))

(deftest a-test
  (let [a (System/currentTimeMillis)]
    (is (= 0 (mod a 3)))))
```

### cider-nrepl version

0.14.0

### Java version

1.8

### Operating system

Mac OS X

**Comment by Malabarba** [comment]
Yes, the debugger needed some special handling in order to support deftest forms, so enlighten will probably require something like that as well.
